### PR TITLE
Create cache and core class representation

### PR DESCRIPTION
**Pull request by katlapinka**

### CODE REVIEW

File: api/cas/core.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        core_info = self._get_core_info()
+        self.core_id = core_info[1]
+        Device.__init__(self, core_info[5])
+        self.cache = cache
+        if self.cache is not None:
+            self.cache.core_devices.append(self)
+
+    def _get_core_info(self):
+        output = TestProperties.executor.execute(
+            list_cmd(OutputFormat.csv.name)).stdout.splitlines()
```
Comment (mmichal10) [inline api/cas/core.py]: Missing command status check.

File: api/cas/cache.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+from test_package.test_properties import TestProperties
+from api.cas.casadm_params import *
+
+
+class Cache:
+    def __init__(self, device_system_path):
+        self.cache_device = Device(device_system_path)
+        self.cache_id = int(self._get_cache_id())
+        self.core_devices = []
+        self._cache_line_size = None
```
Comment (mmichal10) [inline api/cas/cache.py]: Did you mean double underscore? (in both files)

File: api/cas/cache.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+from storage_devices.device import Device
+from test_package.test_properties import TestProperties
+from api.cas.casadm_params import *
+
+
+class Cache:
+    def __init__(self, device_system_path):
+        self.cache_device = Device(device_system_path)
+        self.cache_id = int(self._get_cache_id())
+        self.core_devices = []
```
Comment (mmichal10) [inline api/cas/cache.py]: Why core_devices has no accessor?

File: api/cas/casadm.py [hunk 4]
```diff
@@ -57,25 +44,24 @@ def stop_cache(cache_id: int, no_data_flush: bool = False, shortcut: bool = Fals
...
     _core_id = None if core_id is None else str(id)
     output = TestProperties.executor.execute(
-        add_core_cmd(cache_id=str(cache_id), core_dev=core_dev.system_path,
+        add_core_cmd(cache_id=str(cache.cache_id), core_dev=core_dev.system_path,
                      core_id=_core_id, shortcut=shortcut))
     if output.exit_code != 0:
         raise Exception(
             f"Failed to add core. stdout: {output.stdout} \n stderr :{output.stderr}")
-    return output
+    return Core(core_dev.system_path, cache)
```
Comment (Deixx) [inline api/cas/casadm.py]: cache.cache_id

File: api/cas/core.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    def core_statistics(self,
+                        per_io_class=False,
+                        io_class_id: int = None,
+                        stat_filter: List[StatsFilter] = None,
+                        percentage_val: bool = False):
+        return casadm_parser.get_statistics(self.cache_id, self.core_id, per_io_class, io_class_id,
+                                            stat_filter, percentage_val)
+
+    def status(self):
+        return self._get_core_info()[3]
```
Comment (Deixx) [inline api/cas/core.py]: __get_core_info

File: api/cas/core.py [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+    def seq_cut_off_parameters(self):
+        return casadm_parser.get_seq_cut_off_parameters(self.cache.cache_id, self.core_id)
```
Comment (Deixx) [inline api/cas/core.py]: self.cache_id

File: cas_configuration/cache_config.py [hunk 7]
```diff
@@ -32,15 +33,82 @@ class SeqCutOffPolicy(Enum):
...
+class Time(timedelta):
+    def total_milliseconds(self):
+        return self.total_seconds() * 1000
+
+
+class FlushParametersAlru:
+    def __init__(self):
+        self.activity_threshold = None
+        self.flush_max_buffers = None
+        self.staneless_time = None
```
Comment (Deixx) [inline cas_configuration/cache_config.py]: staleness

File: api/cas/core.py [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+        if output.exit_code != 0:
+            raise Exception("Failed to execute list caches command.")
+        output_lines = output.stdout.splitlines()
+        for line in output_lines:
+            split_line = line.split(',')
+            path = None
+            try:
+                path = self.system_path
+            except Exception:
+                pass
```
Comment (Deixx) [inline api/cas/core.py]: what's going on in this try?

File: test_package/cli/test_cli_start_stop.py [hunk 9]
```diff
@@ -43,25 +44,25 @@ def test_cli_add_remove_default_value(prepare_and_cleanup, shortcut):
...
+    assert len(caches[0].core_devices()) == 1
+    assert caches[0].core_devices()[0].core_device.system_path == core_device.system_path
 
-    casadm.remove_core(1, 1, shortcut=shortcut)
-    caches = casadm.parse_list_caches()
+    casadm.remove_core(cache, 1, shortcut=shortcut)
+    caches = casadm_parser.get_caches()
     assert len(caches) == 1
-    assert len(caches["1"]["cores"]) == 0
+    assert len(caches[0].core_devices()) == 0
```
Comment (Deixx) [inline test_package/cli/test_cli_start_stop.py]: > \>      assert len(caches[0].core_devices()) == 1
> E       TypeError: 'list' object is not callable

Cache has both a list self.core_devices and a method core_devices()

File: api/cas/core.py [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+            list_cmd(OutputFormat.csv.name))
+        if output.exit_code != 0:
+            raise Exception("Failed to execute list caches command.")
+        output_lines = output.stdout.splitlines()
+        for line in output_lines:
+            split_line = line.split(',')
+            path = self.system_path
+            if split_line[0] == "core" and (split_line[2] == self.core_device.system_path
+                                            or split_line[5] == path):
+                return split_line
```
Comment (mmichal10) [inline api/cas/core.py]: How about returning dictionary instead of list of strings?

File: api/cas/casadm_parser.py [hunk 11]
```diff
@@ -130,3 +132,74 @@ def get_statistics(
...
+
+
+def get_cores(cache_id: int):
+    from api.cas.core import Core, CoreStatus
+    cores_list = []
+    lines = casadm.list_caches(OutputFormat.csv).stdout.split('\n')
+    is_proper_core_line = False
+    for line in lines:
+        args = line.split(',')
+        if args[0] == "core" and is_proper_core_line and CoreStatus[args[3].lower()].value[0] <= 1:
```
Comment (mmichal10) [inline api/cas/casadm_parser.py]: When you are parsing raw command output, could you assign it to variable? Now it is pretty hard to read.

File: api/cas/core.py [hunk 12]
```diff
@@ -0,0 +1,83 @@
+#
+# Copyright(c) 2019 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+from typing import List
```
Comment (KamilLepek) [inline api/cas/core.py]: not used

File: api/cas/core.py [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+    empty = 0,
+    active = 1,
+    inactive = 2,
+    detached = 3
+
+
+class Core(Device):
+    def __init__(self, core_device: str, cache_id: int):
+        self.core_device = Device(core_device)
+        self.system_path = None
```
Comment (KamilLepek) [inline api/cas/core.py]: maybe 
```
self.system_path = self.core_device.system_path
self.exported_object_path = None

**Pull request merged**